documentation/API-design-guidelines.md

```diff
@@ -0,0 +1,1396 @@
+# API design guidelines
```

Comment:
POINT 2) of https://github.com/camaraproject/WorkingGroups/issues/256 was addressed by https://github.com/camaraproject/WorkingGroups/pull/257 (and merged), but maybe POINT 1) of the issue (not PR) needs still to be addressed.

@shilpa-padgaonkar do you plan to move the open issues into the new repository?

Reply:
Yes it makes sense to add new issues here. Most issues will progress to PRs and as docs are hosted here, makes sense to do so otherwise would be difficult to map PRs to issues

Reply:
> Make sure that this document is updated with the latest changes contained in [camaraproject/WorkingGroups#256](https://github.com/camaraproject/WorkingGroups/issues/256), or that the PR is closed and re-created in the new project when the document is moved.

I will recreate a PR here for the minor change after this merge.

Reply:
created an issue for myself here https://github.com/camaraproject/Commonalities/issues/19